src/services/data_model_engine.js

```diff
@@ -217,8 +236,18 @@ export default class DataModelEngine {
         await this.entityRepository.storeBundleProofMetadata(waitingBundle.bundleId, blockNumber, timestamp, transactionHash);
         await this.bundleRepository.storeBundleProofMetadata(waitingBundle.bundleId, blockNumber, timestamp, transactionHash);
         summary.ok[waitingBundle.bundleId] = {uploadResult};
+        this.logger.info({
```

Comment:
I don't like that the DataModelEngine is now aware of the logger. We managed to avoid this until now, and I would like to keep it this way. As an alternative I would propose the `uploadAcceptedBundleCandidate` method to accept a optional callback that can be used to report progress. It will be also easier to test (which you should also do)